### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by PrantoshB**
## Hello There 👋

In this branch, I and my coding partner @gauravgangwarweb have:
- Created a class called Solver
- Created a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number
- Created a method called `fizzbuzz` that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
   - When N is divisible by 3, return "fizz".
   - When N is divisible by 5, return "buzz".
   - When N is divisible by 3 and 5, return "fizzbuzz".
   - Any other case, return N as a string (e.g. say N is 7 then return "7").

- Added Test Cases for all these 3 methods.
- Tested using RSpec
- Fixed Linters Errors
- Followed Git and GitHub Flows

### Thanks 🍻

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Solver
+  def factorial(number)
+    if number.negative?
+      'Plese give me a positive number'
```
Comment (Kamba56) [inline solver.rb]: - [ ] As demanded by the project requirements, kindly make this method raise an exception error if the number is negative. Also, test for the error raised.

you can use;
```ruby
if number.negative?
  raise Exception
```

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    context 'should returns the correct reversed word' do
+      it 'returns metsys if word is system' do
+        expect(solver.reverse('system')).to eq('metsys')
+      end
+    end
+  end
+
+  describe '#fizzbuzz' do
+    context 'should returns the correct fizzbuzz' do
+      it 'returns 1 if number is equal to 1' do
```
Comment (Kamba56) [inline spec/solver_spec.rb]: - [Optional] Kindly correct the description of this `it` block. The description should describe what the block is testing. You can use something like;
`it 'Should return number if the number is not divisible by 3 or 5'`.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @PrantoshB, @gauravgangwarweb , 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Git workflow!👏👏👏
- [x] All linters passed
- [x] Great job using TDD to implement the solver class.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PrantoshB** [comment]
Thanks @Kamba56 for the suggestions.

We have implemented the changes and submitting for next review.

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gitbygaurav** [comment]
Hii @Qoosim 👋

Thanks for approving our PR.

**Comment by Qoosim** [comment]
Hey @gauravgangwarweb, you're most welcome brother. :handshake:

**Comment by Kamba56** [comment]
> Thanks @Kamba56 for the suggestions.
> 
> We have implemented the changes and submitting for next review.

You are welcome, happy coding 😁💪🏾

**Pull request merged**